test/e2e/run-test

```diff
@@ -0,0 +1,189 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+SCRIPTPATH="$(cd "$(dirname "$0")"; pwd -P)"
+MAKEFILE="${SCRIPTPATH}/../../Makefile"
+BUILD_DIR="${SCRIPTPATH}/../../build"
+TEST_FAILURES_LOG="${BUILD_DIR}/test-failures.log"
+
+AEIS="${BUILD_DIR}/ec2-instance-selector"
+
+## Build binary for tests to run 
+make -f $MAKEFILE build
```

Comment:
Why build here instead of having build be part of the `make e2e-test` target?

Reply:
that's a good point. I wanted it to be standalone executable, but I suppose if the binary doesn't exist, it would just error. If it was out-of-date though, could get bad results.

Reply:
any particular reason why you want it to be standalone executable?

Reply:
I think the only reason would be if they grow and we'd need to add some CLI options to fail fast or do some other optimization when running them.   Was interested in using local-stack (they don't currently support DescribeInstanceTypes or DescribeInstanceTypeOfferings, but maybe in the future) so that they could be run without credentials, which could be a CLI option. Make doesn't make it easy to pass options like that.  But I think simplifying it now to only be run through make is fine. If we end up adding stuff later, we can reevaluate, or do a more clever check on the binary to make sure the latest version is being executed.